telemetry/main.schema.json

```diff
@@ -846,13 +846,16 @@
           "minimum": 0
         },
         {
-          "type": "string"
+          "type": "string",
+          "pattern": "^[a-zA-Z][a-zA-Z0-9_.]*[a-zA-Z0-9]$"
```

Comment:
@fedepad , this slightly changed meanwhile :) We're now using templatized schemas that get build off the schema sections in templates/include/telemetry. We don't need to change multiple schemas anymore, but making this change has become slightly more involved. If you're still interested in working on it, at the very least you should:

-  rebase the patch and move your changes to the schema to the [template file](https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/master/templates/include/telemetry/processData.1.schema.json#L50);
- create a new file called _pattern_event_string.1.schema.json_ with a content similar to [this one](https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/master/templates/include/common/pattern_uuid.1.schema.json), but with your regex;
- reference the pattern in your schema change as done [here](https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/master/templates/include/telemetry/clientId.1.schema.json#L3)

@trink do we need to build and commit the output schemas too (e.g. schemas/telemetry/*)?